src/source/raster_tile_source.js

```diff
@@ -238,7 +238,20 @@ class RasterTileSource extends Evented implements Source {
 
     // $FlowFixMe[method-unbinding]
     unloadTile(tile: Tile, callback: Callback<void>) {
-        if (tile.texture) this.map.painter.saveTileTexture(tile.texture);
+        // Cache the tile texture to avoid re-allocating Textures if they'll just be reloaded
+        if (tile.texture && tile.texture instanceof Texture) {
+            // Clean everything else up owned by the tile, but preserve the texture.
+            // Destroy first to prevent racing with the texture cache being popped.
+            tile.destroy(true);
+
+            // Save the texture to the cache
+            if (tile.texture && tile.texture instanceof Texture) {
```

Comment:
Nit: redundant check

Reply:
I thought that too! Seems that Flow picks up that `tile` may have been modified

Reply:
In that case you can usually placate Flow by separating the condition (or a texture reference) into a variable.